proto/allocation.proto

```diff
@@ -0,0 +1,140 @@
+syntax = "proto3";
+
+package smartcore.bos;
+
+option go_package = "github.com/smart-core-os/sc-bos/pkg/gen";
+
+import "google/protobuf/field_mask.proto";
+import "google/protobuf/timestamp.proto";
+import "types/time/period.proto";
+import "actor.proto";
+
+// AllocationApi characterizes the availability of a resource that may be allocated to an actor for a period of time.
+// Examples include meeting rooms, desks, parking spots, or shared equipment such as lockers.
+service AllocationApi {
+  // Get the current allocation for a resource.
+  rpc GetAllocation(GetAllocationRequest) returns (Allocation);
+  // Create or update an allocation for a resource.
+  rpc UpdateAllocation(UpdateAllocationRequest) returns (Allocation);
+  // Subscribe to changes in allocations for a resource.
+  rpc PullAllocations(PullAllocationsRequest) returns (stream PullAllocationsResponse);
+  // Lists all allocatable resources and their current allocations.
+  rpc ListAllocatableResources(ListAllocatableResourcesRequest) returns (ListAllocatableResourcesResponse);
+}
```

Comment:
I know this PR has merged already, but I'm confused by the structure of this trait. Why does ListAllocatableResources follow a different naming structure from all the other methods (and all the other traits)?

As a second question, do we need to add lists of allocations at this stage? Not against it, but if we don't need it then it would make things a lot simpler in both concept and implementation.